### PR TITLE
Feat(designer): test chat fixes/enhancements

### PR DESCRIPTION
**Pull request by rachradulo**
### **PR Type**
Enhancement, Bug fix

___

### **Description**
- Implement thinking budget enforcement via logits processor for streaming

- Clean up generation settings UI to only show functioning parameters

- Persist model, database, and strategy selections in test chat

- Improve scroll UX with sticky scroll and jump-to-latest button

- Auto-collapse thinking steps by default with user toggle capability

- Fix thinking budget and max tokens parameter handling across clients

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Thinking Budget<br/>Enforcement"] -->|"Logits Processor"| B["GGUF Language<br/>Model"]
  C["Generation<br/>Settings"] -->|"Cleaned UI"| D["Test Chat<br/>Component"]
  E["Model/DB/Strategy<br/>Selection"] -->|"localStorage"| D
  F["Scroll UX<br/>Improvements"] -->|"Jump Button<br/>Auto-scroll"| D
  G["Parameter<br/>Extraction"] -->|"extra_body"| H["OpenAI &<br/>Ollama Clients"]
```

<details><summary><h3>File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>7 files</summary><table>
<tr>
  <td><strong>gguf_language_model.py</strong><dd><code>Enforce thinking budget via logits processor</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/609/files#diff-aa3d47952f93f3f6d338873bbc86b4f9eb2700b79e17fd3e73e5a2016fc37064">+15/-16</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>thinking.py</strong><dd><code>Refactor ThinkingBudgetProcessor for numpy arrays</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/609/files#diff-776fbb1ef225969752ae4738778c2af310770e5d6fa91372fb798cf654638f90">+38/-22</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>ollama.py</strong><dd><code>Support max_tokens via extra_body options</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/609/files#diff-793dea6a5bda9526672d577279a5fbe5a60aa282044ff10b8d4dcb397b7cc32a">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>projects.py</strong><dd><code>Pass max_tokens parameter to chat service</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/609/files#diff-ab4cacdb0344ceca710f889b0d839b684ea3bd780df3f6125661cf30950b287a">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>project_chat_service.py</strong><dd><code>Include max_tokens in extra_body for chat</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/609/files#diff-50584d8236197d5cd0b1c36323fabc7535112d71414a8e2870ee4ae4817e63c6">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Test.tsx</strong><dd><code>Clean up generation settings and add click-outside close</code>&nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/609/files#diff-aa14e4524fb23a4b4085f4603c9d4886f2ae4fc5c3e8427d42a2bfab284918f5">+154/-190</a></td>

</tr>

<tr>
  <td><strong>TestChat.tsx</strong><dd><code>Persist selections, improve scroll UX, auto-collapse thinking</code></dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/609/files#diff-2b8499a6afdd702c2434e7b6914dda681187f8e169f1265c7d9ab18ac7d774f0">+299/-143</a></td>

</tr>
</table></details></td></tr><tr><td><strong>Bug fix</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>service.py</strong><dd><code>Extract thinking params from extra_body fallback</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/609/files#diff-0d5bb320d0590e4a8fb439fd35b07be371bae7d283b4862979f25fbaad05be14">+18/-3</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>openai.py</strong><dd><code>Extract max_tokens from extra_body to api_params</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/609/files#diff-202f6c8c05ab4d3c6ca7a060d7d8fbcb362d16361dd1fe03a062c5551c5a0058">+18/-2</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tbody></table>

</details>

___

### CODE REVIEW

**Review by mhamann — APPROVED** [review]

**Review by BobbyRadford — COMMENTED** [review]
lgtm
